src/ir/function.rs

```diff
@@ -186,7 +186,8 @@ impl FunctionSig {
             }
         }
 
-        let ret = try!(Item::from_ty(&ty.ret_type(), None, None, ctx));
+        let ty_ret_type = ty.ret_type().expect("Expected a valid return type");
```

Comment:
Thanks you.
I'll update the code.
 - Should we add a test for this as well?
 - I was under the impression that this code path should not happen (modulo bugs), do you have an idea of what C++ code would trigger it?